main.py

```diff
@@ -183,156 +184,154 @@ def compose_message(issues):
+        5. return them
+        """
+        return random.choices(issues, k=number_of_issues)
+
+
+def choose_issues_queue(issues, number_of_issues, author=None):
+    chosen_issues=[]
+
+    if author:
+        issues = pr_with_pending_review(get_pr_details(issues), author)
+        # since async islice doesn't exist
+        async for issue in issues:
+            chosen_issues.append(issue)
+            if len(chosen_issues) == number_of_issues:
+                break
+    else:
+        async for issue in issues:
+            chosen_issues.append(issue)
+            if len(chosen_issues) == number_of_issues:
+                break
 
-        chosen_issues = random.choices(issues, k=number_of_issues)
-        reply.description = compose_message(beautify_issues(chosen_issues))
-        if len(chosen_issues) < number_of_issues:
-            reply.set_footer(text="There wasn't enough...")
-    await channel.send(embed=reply)
-    await set_playing('The Waiting Game')
+    return chosen_issues
 
 
-async def review_q(channel, number_of_issues, author=None):
+async def queue_handler(channel, number_of_issues, author=None, git_config):    
     await set_playing('On The Cue')
     error_channel = error_handler(channel)
     async with channel.typing():
-        issues = get_issues(config["repo"], pull_request=True,
-                            exclude_labels=['status: stale'],
-                            include_labels=['needs: code review'],
-                            sort='updated', ascending_order=True,
-                            error_channel=error_channel)
+        issues = get_issues(**git_config)
         reply = discord.Embed(color=discord.Color.purple())
 
-        if author:
-            title_suffix = f" waiting @{author}'s review:"
-            issues = pr_with_pending_review(get_pr_details(issues), author)
-            chosen_issues = []
-            # since async islice doesn't exist
-            async for issue in issues:
-                chosen_issues.append(issue)
-                if len(chosen_issues) == number_of_issues:
-                    break
-        else:
-            title_suffix = " in review queue:"
-            chosen_issues = []
-            async for issue in issues:
-                chosen_issues.append(issue)
-                if len(chosen_issues) == number_of_issues:
-                    break
-
-        reply.title = f"Oldest {number_of_issues} PR" + title_suffix
+        if git_config["command"]=="rand":
+            chosen_issues=choose_issues_rand(issues, number_of_issues)
+            reply.title = f"{number_of_issues} random picks out of {len(issues)}:"
+        elif git_config["command"]=="review" or git_config["command"]=="feedback":
+            chosen_issues=choose_issues_queue(issues, number_of_issues, author)
+            reply.title = f"Oldest {number_of_issues} " + ("PR" if git_config["pull_request"] else "issues") + (f' for @{author}:' if author else ' in queue:')
```

Comment:
Do suggest a better alternative, if possible

Reply:
```suggestion
           kind = "PR" if git_config["pull_request"] else "issues"
           selection = f for @{author}" if author else "in queue"
           reply.title = f"Oldest {number_of_issues} {kind} {selection}:"
```
When in doubt, divide

Reply:
What about the function `choose_issues_rand`, `choose_issues_queue` and their calls?

Reply:
I'll take another look sometime soon